### PR TITLE
Set up postgres

### PR DESCRIPTION
**Pull request by IzzyLeeDev**
Hey @IamTechknow 

Could you check out this It's a biggish one, but lots of it is in first-draft form. I need to work on submitting smaller, more frequent PRs! Hopefully this comes in time to have a more easily clonable service for all your different proxies, with at least one route that hooks into the Postgres RESTful API. 

Thanks team!

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -1,129 +1,119 @@
...
+  const newComment = {
+    id: req.body.id,
+    author: req.body.author,
+    authorIsCreator: req.body.authorIsCreator, // removed in future data generations
+    profilePicture: req.body.profilePicture,
+    createdAt: req.body.createdAt,
+    body: req.body.body,
+    replies: [],
+  };
+  Models.createComment(req.params.projectId, newComment, (dbErr, dbRes) => {
```
Comment (IamTechknow) [inline server/server.js:65]: If you pass in req.body directly, then you don't need the newComment declaration

File: server/server.js [hunk 2]
```diff
@@ -1,129 +1,119 @@
...
+  const newComment = {
+    id: req.body.id,
+    author: req.body.author,
+    authorIsCreator: req.body.authorIsCreator, // removed in future data generations
+    profilePicture: req.body.profilePicture,
+    createdAt: req.body.createdAt,
+    body: req.body.body,
+    replies: [],
+  };
+  Models.createReply(req.params.projectId, req.params.commentId, newComment, (dbErr, dbRes) => {
```
Comment (IamTechknow) [inline server/server.js:86]: Same as above comment. You can extend req.body to add the replies property

File: seedPostgres.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    .catch((err) => console.error(err) );
+  db.query(`COPY projects(_id, author, comments) FROM '${csvData}' CSV HEADER;`)
+    .catch((err) => console.error(err) );
+
+  console.log(`Seeded campaigns in ${new Date().getTime() - startTime} ms`);
+
+  db.end();
+}
+
+seedPostgresData();
```
Comment (IamTechknow) [inline seedPostgres.js:27]: You may use async/await here to turn the promise based code to synchronous code, thus ensuring the code flows in the order you probably intended. I recommend you look it up!

File: dataGenerator.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+      break;
+    } else {
+     // console.log('writing', i, '!');
+    }
+    if (i >= projectsToMake) {
+        stream.write(']');
+        bracketWritten = true;
+        break;
+    }
+  } if(i === projectsToMake) {
```
Comment (IamTechknow) [inline dataGenerator.js:122]: No newline separation between while loop and if statement

File: models/PostgresModels.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      newComment.author,
+      newComment.replies,
+      newComment.createdAt,
+      newComment.profilePicture
+    ],
+    callback
+  );
+};
+
+// TODO:-----------------------------------------
```
Comment (IamTechknow) [inline models/PostgresModels.js:34]: This is commented code, please uncomment or delete!

**Review by IamTechknow — CHANGES_REQUESTED** [review]
In general your code is fine, there is a bit of redundancy. I did notice you combined React code with Postgres setup. In Windows you may use the git gui to commit certain files and lines to fine tune your commits.

File: server/server.js (hunk 1 above)
Comment (IzzyLeeDev) [reply server/server.js:65]: Thanks! change made.

**Pull request merged**